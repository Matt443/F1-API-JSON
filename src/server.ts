import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}