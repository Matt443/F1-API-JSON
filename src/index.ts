import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
