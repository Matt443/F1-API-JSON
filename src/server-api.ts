import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
