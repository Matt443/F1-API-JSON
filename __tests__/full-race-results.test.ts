import { getFullRaceResults } from "../src/server";

test("get race results", async () => {
    expect(await getFullRaceResults()).toMatchSnapshot();
});
