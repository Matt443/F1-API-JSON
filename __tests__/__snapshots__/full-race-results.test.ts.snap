// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get race results 1`] = `
[
  {
    "driverCode": "NOR",
    "laps": 57,
    "name": "Lando Norris",
    "number": 4,
    "points": 25,
    "team": "McLaren Mercedes",
    "time": "1:42:06.304",
  },
  {
    "driverCode": "VER",
    "laps": 57,
    "name": "Max Verstappen",
    "number": 1,
    "points": 18,
    "team": "Red Bull Racing Honda RBPT",
    "time": "+0.895s",
  },
  {
    "driverCode": "RUS",
    "laps": 57,
    "name": "George Russell",
    "number": 63,
    "points": 15,
    "team": "Mercedes",
    "time": "+8.481s",
  },
  {
    "driverCode": "ANT",
    "laps": 57,
    "name": "Kimi Antonelli",
    "number": 12,
    "points": 12,
    "team": "Mercedes",
    "time": "+10.135s",
  },
  {
    "driverCode": "ALB",
    "laps": 57,
    "name": "Alexander Albon",
    "number": 23,
    "points": 10,
    "team": "Williams Mercedes",
    "time": "+12.773s",
  },
  {
    "driverCode": "STR",
    "laps": 57,
    "name": "Lance Stroll",
    "number": 18,
    "points": 8,
    "team": "Aston Martin Aramco Mercedes",
    "time": "+17.413s",
  },
  {
    "driverCode": "HUL",
    "laps": 57,
    "name": "Nico Hulkenberg",
    "number": 27,
    "points": 6,
    "team": "Kick Sauber Ferrari",
    "time": "+18.423s",
  },
  {
    "driverCode": "LEC",
    "laps": 57,
    "name": "Charles Leclerc",
    "number": 16,
    "points": 4,
    "team": "Ferrari",
    "time": "+19.826s",
  },
  {
    "driverCode": "PIA",
    "laps": 57,
    "name": "Oscar Piastri",
    "number": 81,
    "points": 2,
    "team": "McLaren Mercedes",
    "time": "+20.448s",
  },
  {
    "driverCode": "HAM",
    "laps": 57,
    "name": "Lewis Hamilton",
    "number": 44,
    "points": 1,
    "team": "Ferrari",
    "time": "+22.473s",
  },
  {
    "driverCode": "GAS",
    "laps": 57,
    "name": "Pierre Gasly",
    "number": 10,
    "points": 0,
    "team": "Alpine Renault",
    "time": "+26.502s",
  },
  {
    "driverCode": "TSU",
    "laps": 57,
    "name": "Yuki Tsunoda",
    "number": 22,
    "points": 0,
    "team": "Racing Bulls Honda RBPT",
    "time": "+29.884s",
  },
  {
    "driverCode": "OCO",
    "laps": 57,
    "name": "Esteban Ocon",
    "number": 31,
    "points": 0,
    "team": "Haas Ferrari",
    "time": "+33.161s",
  },
  {
    "driverCode": "BEA",
    "laps": 57,
    "name": "Oliver Bearman",
    "number": 87,
    "points": 0,
    "team": "Haas Ferrari",
    "time": "+40.351s",
  },
  {
    "driverCode": "LAW",
    "laps": 46,
    "name": "Liam Lawson",
    "number": 30,
    "points": 0,
    "team": "Red Bull Racing Honda RBPT",
    "time": "DNF",
  },
  {
    "driverCode": "BOR",
    "laps": 45,
    "name": "Gabriel Bortoleto",
    "number": 5,
    "points": 0,
    "team": "Kick Sauber Ferrari",
    "time": "DNF",
  },
  {
    "driverCode": "ALO",
    "laps": 32,
    "name": "Fernando Alonso",
    "number": 14,
    "points": 0,
    "team": "Aston Martin Aramco Mercedes",
    "time": "DNF",
  },
  {
    "driverCode": "SAI",
    "laps": 0,
    "name": "Carlos Sainz",
    "number": 55,
    "points": 0,
    "team": "Williams Mercedes",
    "time": "DNF",
  },
  {
    "driverCode": "DOO",
    "laps": 0,
    "name": "Jack Doohan",
    "number": 7,
    "points": 0,
    "team": "Alpine Renault",
    "time": "DNF",
  },
  {
    "driverCode": "HAD",
    "laps": 0,
    "name": "Isack Hadjar",
    "number": 6,
    "points": 0,
    "team": "Racing Bulls Honda RBPT",
    "time": "DNS",
  },
]
`;
